transaction.go

```diff
@@ -94,56 +203,83 @@ func cbPAMConv(s C.int, msg *C.char, c C.uintptr_t) (*C.char, C.int) {
 	var err error
 	v := cgo.Handle(c).Value()
 	style := Style(s)
+	var handler ConversationHandler
 	switch cb := v.(type) {
 	case BinaryConversationHandler:
 		if style == BinaryPrompt {
 			bytes, err := cb.RespondPAMBinary(BinaryPointer(msg))
 			if err != nil {
-				return nil, C.PAM_CONV_ERR
+				return nil, ErrConv.toC()
```

Comment:
I would prefer `C.int(err)`, I think the `toC` method obscures the code a bit.

Reply:
IIRC the reason for it to exist was to be able to use it also in test, but that can be the only usage maybe.